### PR TITLE
Query and update

### PR DESCRIPTION
**Pull request by Bellagirl-maker**
-Add a column species of type string to animals table and modify  schema.sql file.
-Insert data in data.sql file
-Update and delete data inside a transaction.
- Write queries to answer questions from animals table.
- Add screenshots
Delete 1
Delete 2
Delete 3
q1
q2
transaction update 1
transaction update 2
update  3
update 1
update 2

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -6,3 +6,18 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
 SELECT * FROM animals WHERE name <> 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT del_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO SAVEPOINT del_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
```
Comment (Ghiftee) [inline queries.sql:33]: - [ ] Good job adding this transaction and screenshots, however, you are missing a few transactions before this. Kindly find them circled in the image below and add written queries for themthem.
Screenshot 2023-06-13 at 15 20 10 (2)

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @Bellagirl-maker 👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ Screenshots are added
✔️ Queries are successful
✔️ Transactions are added

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: queries.sql [hunk 2]
```diff
@@ -6,3 +6,28 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name <> 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- update the animals table by setting the species column to unspecified
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
```
Comment (Ghiftee) [inline queries.sql]: - [ ] According to the requirements, after making this change, kindly verify that the change was made before rollback and afterwards verify that the changes were undone e.g.

```
BEGIN; -- start transaction

UPDATE animals
SET species = 'unspecified'; -- make change

SELECT species from animals; -- verify that change was made
 ROLLBACK;

 SELECT species from animals; -- verify that change was undone
```

File: queries.sql [hunk 3]
```diff
@@ -6,3 +6,28 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon
+UPDATE animals SET species = 'unspecified' WHERE name LIKE '%mon';
```
Comment (Ghiftee) [inline queries.sql]: - [ ] Here, on line 15, you set the species to `unspecified`, however, you are required to set it to `digimon`, kindly fix this

File: queries.sql [hunk 4]
```diff
@@ -6,3 +6,28 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
```
Comment (Ghiftee) [inline queries.sql]: - [ ] After writing this query, you are required to verify that the change was made, commit it and verify it persists after the commit

```sql
 SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```

File: queries.sql [hunk 5]
```diff
@@ -6,3 +6,28 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+
```
Comment (Ghiftee) [inline queries.sql:26]: - [ ] You are still missing one transaction as shown below. Please add it
Screenshot 2023-06-13 at 15 20 10 (2)

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hi there @Bellagirl-maker 👋🏽 , 

Great work on making the changes requested previously👏 
To highlight 👍🏽 
✔️ Missing transactions were added
✔️ Screenshots were updated

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 4 above)
Comment (Bellagirl-maker) [reply queries.sql]: @Ghiftee done with this change

File: queries.sql (hunk 3 above)
Comment (Bellagirl-maker) [reply queries.sql]: @Ghiftee done with this change

File: queries.sql (hunk 2 above)
Comment (Bellagirl-maker) [reply queries.sql]: @Ghiftee done with this change

**Review by topeogunleye — APPROVED** [review]
### Hi @Bellagirl-maker, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Verified that the change was made before rollback. ✔️ 
-  Set the species from unspecified to digimon ✔️ 
- Added missing transaction ✔️ 
 
## [Optional] suggestions:
* [ ]  For future pull requests, please notice that your branch name `query-and-update` does not follow the convention of being named after a feature or action. For example, you could name your branch `added-query-and-update`. This makes it easier to track the changes and the purpose of each branch.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**